plugins/quickcl.py

```diff
@@ -241,67 +249,45 @@ def __init__(self, microscope, main_app):
 
         self.addMenu("Acquisition/Quick CL...\tF2", self.start)
 
-    def _get_new_filename(self):
-        conf = get_acqui_conf()
-        return os.path.join(
-            conf.last_path,
-            u"cl-%s-0000.png" % (time.strftime("%Y%m%d-%H%M%S"),)
-        )
-
-    def _get_filename_count(self, fn):
-        """
-        Find a number pattern at the end of the filename (non including the extension)
 
-        return (str): the pattern
-        """
-        bn, ext = os.path.splitext(fn)
-        bn_ncnt = bn.rstrip(string.digits)
-        logging.debug("bn = %s, ncnt = %s", bn, bn_ncnt)
-        return bn[len(bn_ncnt):]
-
-    def _set_filename(self, fn):
+    def _get_new_filename(self):
         """
-        Setter, which ensures there is always a -0000 pattern at the end of the filename
+        Get filename from pattern in conf file
         """
-        # Find a pattern, or add one if none present
-        cnt = self._get_filename_count(fn)
-        logging.debug("cnt = %s", cnt)
-        if not cnt:
-            bn, ext = os.path.splitext(fn)
-            fn = bn + "-0000" + ext
-            logging.debug("Updating filename to %s", fn)
-
+        # Make sure count has 4 digits
+        # self.conf.fn_count = '%04s' % self.conf.fn_count
+        self.conf.last_extension = '.png'
+        fn = create_filename(self.conf.last_path, self.conf.fn_ptn,
+                                       self.conf.last_extension, self.conf.fn_count)
+        self.conf.fn_count = update_counter(self.conf.fn_count)
         return fn
 
     def _on_filename(self, fn):
+        """
+        Warn if extension not .png, store path and pattern in conf file
+        """
         bn, ext = os.path.splitext(fn)
         if not ext.endswith(".png") and not ALLOW_SAVE:
             logging.warning("Only PNG format is recommended to use")
 
         # Store the directory so that next filename is in the same place
-        conf = get_acqui_conf()
         p, bn = os.path.split(fn)
         if p:
-            conf.last_path = p
+            self.conf.last_path = p
 
-        logging.debug("Filename is now %s", self.filename.value)
+        # Save pattern
+        self.conf.fn_ptn, self.conf.fn_count = guess_pattern(fn)
 
-    def _bump_filename_number(self):
-        """
-        Find the digit pattern in the filename and replace by +1
-        """
-        fn = self.filename.value
-        bn, ext = os.path.splitext(fn)
-        num_str = self._get_filename_count(fn)
-        assert num_str, "No number in the filename"
-        n = int(num_str) + 1
-        fmt_str = ("%%0%dd" % len(num_str))
-        num_str = fmt_str % n
-        fn = bn[:-len(num_str)] + num_str + ext
+        # Save last format
+        fmt = importlib.import_module(ext, "odemis.dataio").FORMAT
```

Comment:
No need to store the format. Most importantly you need to twist your tong 7 times before using `importlib`. It's a pretty messy/dangerous module. In that particular case, it would go well if the user picks any extension which doesn't happen to also be the name of a module.